include/engine/polyline_compressor.hpp

```diff
@@ -10,21 +10,15 @@ namespace osrm
 {
 namespace engine
 {
-namespace detail
-{
-constexpr double POLYLINE_PRECISION = 1e5;
-constexpr double COORDINATE_TO_POLYLINE = POLYLINE_PRECISION / COORDINATE_PRECISION;
-constexpr double POLYLINE_TO_COORDINATE = COORDINATE_PRECISION / POLYLINE_PRECISION;
-}
 
 using CoordVectorForwardIter = std::vector<util::Coordinate>::const_iterator;
 // Encodes geometry into polyline format.
 // See: https://developers.google.com/maps/documentation/utilities/polylinealgorithm
-std::string encodePolyline(CoordVectorForwardIter begin, CoordVectorForwardIter end);
+std::string encodePolyline(CoordVectorForwardIter begin, CoordVectorForwardIter end, double polylinePrecision = 1e5);
```

Comment:
Same goes for here, a template parameter would help the optimizer to pre-compute some arithmetic values for us. Example:
```
template<unsigned PRECISION=1e5>
std::string encodePolyline(CoordVectorForwardIter begin, CoordVectorForwardIter end)
{
   /* implementation goes here */
}
```